instrumentation/action_mailer/lib/opentelemetry/instrumentation/action_mailer/instrumentation.rb

```diff
@@ -0,0 +1,82 @@
+      # The Instrumentation class contains logic to detect and install the ActionMailer instrumentation
+      class Instrumentation < OpenTelemetry::Instrumentation::Base
+        MINIMUM_VERSION = Gem::Version.new('6.1.0')
+        install do |_config|
+          resolve_email_address
+          ecs_mail_convention
+          require_dependencies
+        end
+
+        present do
+          defined?(::ActionMailer)
+        end
+
+        compatible do
+          gem_version >= MINIMUM_VERSION
+        end
+
+        option :disallowed_notification_payload_keys, default: [], validate: :array
+        option :notification_payload_transform,       default: nil, validate: :callable
+        option :email_address,                        default: :omit, validate: %I[omit include]
+
+        private
+
+        def gem_version
+          ::ActionMailer.version
+        end
+
+        def resolve_email_address
+          return unless ActionMailer::Instrumentation.instance.config[:email_address] == :omit
+
+          ActionMailer::Instrumentation.instance.config[:disallowed_notification_payload_keys] += ['email.to.address', 'email.from.address', 'email.cc.address', 'email.bcc.address']
+        end
+
+        def ecs_mail_convention
+          if ActionMailer::Instrumentation.instance.config[:notification_payload_transform].nil?
+            transform_attributes = ->(payload) { transform_payload(payload) }
+          else
+            original_callable = ActionMailer::Instrumentation.instance.config[:notification_payload_transform]
+            transform_attributes = lambda do |payload|
+              new_payload = transform_payload(payload)
+              user_payload = original_callable.call(new_payload)
+              if user_payload.instance_of?(Hash)
+                user_payload
+              else
+                OpenTelemetry.logger.error("ActionMailer: transformed payload is #{user_payload.class} (require Hash)")
+                new_payload
+              end
+            end
+          end
+          ActionMailer::Instrumentation.instance.config[:notification_payload_transform] = transform_attributes
+        end
+
+        # email attribute key convention is obtained from: https://www.elastic.co/guide/en/ecs/8.11/ecs-email.html
+        def transform_payload(payload)
+          new_payload = {}
+          new_payload['email.message_id'] = payload[:message_id] if payload[:message_id]
+          new_payload['email.subject'] = payload[:subject] if payload[:subject]
+          new_payload['email.x_mailer'] = payload[:mailer] if payload[:mailer]
+          new_payload['email.to.address'] = payload[:to] if payload[:to]
+          new_payload['email.from.address'] = payload[:from] if payload[:from]
+          new_payload['email.cc.address'] = payload[:cc] if payload[:cc]
+          new_payload['email.bcc.address'] = payload[:bcc] if payload[:bcc]
+          new_payload['email.origination_timestamp'] = payload[:date] if payload[:date]
```

Comment:
It looks like the original payload used symbols as keys, but the new payload is using strings. Is that intended? Would there be a performance benefit by making the strings symbols, ex. `:'email.message_id'`?

Reply:
These are used to populate the span so they must be strings as spans do not allow symbol key attributes

Reply:
Got it, thanks @arielvalentin! What about making them constants?

Reply:
To avoid "magic variables"?

Reply:
I was thinking string allocations, but I don't think this is going to be run very many times, so I don't think we need to worry about it here.